lib/src/position.dart

```diff
@@ -243,6 +291,16 @@ abstract class Position<T extends Position<T>> {
   bool isLegal(Move move) {
     assert(move is NormalMove || move is DropMove);
     if (move is NormalMove) {
+      if (this is ChessSharp) {
```

Comment:
For the sake of code readability I wonder if we can instead override `isLegal` in `ChessSharp` like so:

```dart
@override
bool isLegal(Move move) {
  if (move is NormalMove) {
          // In Chess♯, pieces on the board cannot move until the king has been placed
         if (pockets!.of(turn, Role.king) > 0) {
           return false;
         }
         // In Chess♯, pawns may only promote to queens
         if (move.promotion != null && move.promotion != Role.queen) {
           return false;
         }
     } else if (move is DropMove) {
             if (move.role == Role.queen &&
           pockets!.countSide(turn) > 1) {
         return false;
     }
     return super.isLegal(move);
  }
```

I may be wrong, but it seems to me that extra logic of chess sharp can be checked first and then normal chess logic applies.